### PR TITLE
Adds 1m+ prefix test

### PR DESCRIPTION
**Pull request by jjacob512**
Signed-off-by: Jacob M. Jacob <4000419+jjacob512@users.noreply.github.com>

### CODE REVIEW

File: tests/functional/meson.build [hunk 1]
```diff
@@ -15,6 +15,10 @@ tests = {
         'suites': ['python', 'short'],
         'depends': hse_python_depends,
     },
+    'kvs_many_pfx': {
+        'suites': ['python', 'long'],
```
Comment (tristan957) [inline tests/functional/meson.build:19]: On one of our lab machines with an nvme drive, can you quantify what how long it takes?

File: tests/functional/kvs_many_pfx.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+from utility import lifecycle, cli
+
+# creates 1million + prefixes , adds data to each prefix and validates.
+
+# Test params
+# total number of prefixes
+NUM_PFXS = 1024 * 1024 + 1024
+# number of keys per prefix
+NUM_KEYS = 10
```
Comment (tristan957) [inline tests/functional/kvs_many_pfx.py:19]: can you make sure this file is formatted with black?

File: tests/functional/kvs_many_pfx.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+hse.init(cli.CONFIG)
+
+
+def create_pfx_data(kvdb, kvs, pfx):
```
Comment (tristan957) [inline tests/functional/kvs_many_pfx.py:24]: Can you add type annotations?

File: tests/functional/meson.build (hunk 1 above)
Comment (jjacob512) [reply tests/functional/meson.build:19]: 79 secs on an nvme drive.

File: tests/functional/kvs_many_pfx.py (hunk 2 above)
Comment (jjacob512) [reply tests/functional/kvs_many_pfx.py:19]: yes, already did.

File: tests/functional/kvs_many_pfx.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    pfx_bytes = pfx.to_bytes(4, byteorder="big")
+
+    with kvdb.transaction() as txn:
+        for sfx in range(NUM_KEYS):
+            key = pfx_bytes + sfx.to_bytes(1, byteorder="big")
+            kvs.put(key, key, txn=txn)
+
+
+def verify_pfx_data(kvdb, kvs, pfx):
+    pfx_bytes = pfx.to_bytes(4, byteorder="big")
```
Comment (tristan957) [inline tests/functional/kvs_many_pfx.py:34]: Does byteorder matter here? Can we not leave it up to the system default?

Comment (jjacob512) [reply tests/functional/kvs_many_pfx.py:34]: the byte order is big so that int-order matches lexical-order.

**Review by alexttx — APPROVED** [review]

File: tests/functional/meson.build (hunk 1 above)
Comment (tristan957) [reply tests/functional/meson.build:19]: Does this need to be marked as long then? How long does it take on a github runner (Bhavesh can you a machine to test)? Should this also be marked as `functional`?

File: tests/functional/kvs_many_pfx.py (hunk 2 above)
Comment (tristan957) [reply tests/functional/kvs_many_pfx.py:19]: Seems fishy, but if you said you did, then all is good.

**Review by tristan957 — APPROVED** [review]

File: tests/functional/meson.build (hunk 1 above)
Comment (bhaveshvasandani) [reply tests/functional/meson.build:19]: Yes, functional would be the right suite for this test

Comment (jjacob512) [reply tests/functional/meson.build:19]: Here are other tests in the long suite for comparison. My suggestion is to leave it as is.. as this is not the fastest in the suite.

1/3 hse:functional+python+long / kvs_scan_a_million              OK                3.42s
2/3 hse:functional+python+long / kvs_many_pfx                    OK               81.25s
3/3 hse:functional+python+long / kvs_compare_full_and_pfx        OK              118.57s

Comment (bhaveshvasandani) [reply tests/functional/meson.build:19]: Jacob, do you not want this test to run in the CI?

Comment (jjacob512) [reply tests/functional/meson.build:19]: ideally, nightly would be good enough.

**Pull request merged**